starlite/middleware/session/memory_backend.py

```diff
@@ -0,0 +1,53 @@
+from datetime import datetime, timedelta
+from typing import Dict, Optional, Type
+
+from starlite.middleware.session.base import ServerSideBackend, ServerSideSessionConfig
+
+
+class MemoryBackend(ServerSideBackend["MemoryBackendConfig"]):
+    def __init__(self, config: "MemoryBackendConfig") -> None:
+        """Session backend to store data in memory.
+
+        Warning:
+            This should not be used in production and serves mainly as a dummy backend
+            for easy testing. It is not thread or process-safe, and data won't be persisted
+        """
+        super().__init__(config=config)
+        self._store: Dict[str, tuple[datetime, bytes]] = {}
```

Comment:
why not use an instance of the `SimpleCacheBacjend` or extend it? It gives you the functionalities already and implements async locking, so you do get thread safety.

Reply:
I didn't want to create a possible future maintenance dependency here, but sure, there's no reason not to aside from that.

Reply:
Alright, it's simple enough that it doesn't matter. I'll use that.